dogowner/models.py

```diff
@@ -0,0 +1,48 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+class Gender(models.TextChoices):
+    Male = 'M', 'Male'
+    Female = 'F', 'Female'
+    Unknown = 'UN', 'Unknown'
+
+
+class Dogowner(models.Model):
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    first_name = models.CharField(max_length=15, blank=True)
+    last_name = models.CharField(max_length=15, blank=True)
+    phone_number = models.IntegerField(null=True, blank=True)
+    dog_name = models.CharField(max_length=15, blank=True)
+    dog_race = models.CharField(max_length=15, blank=True)
+    dog_picture_url = models.CharField(max_length=1000, blank=True)
+    dog_age = models.IntegerField(null=True, blank=True)
+    dog_weight = models.FloatField(null=True, blank=True)
+    dog_gender = models.CharField(
+        max_length=2,
+        choices=Gender.choices,
+        default='UN', blank=True)
+
+    def __str__(self):
+        return self.first_name + self.last_name
+
+
+@receiver(post_save, sender=User)
+def create_user_profile(sender, instance, created, **kwargs):
```

Comment:
so the"@reciver" was the magic solution for this? surely this will help us all

Reply:
Exactly, we use signal of post save for creation&saving Dog-owner after signal occurred.

Reply:
As I see it we have 3 options :
1) Create a function for saving 'User' with our requierd field and than saving 'Dog-Owner' fields - Omer told me it is not the right way for doing it.
2) Init 'User' in dog-owner creation, but than in tests i have to save/delete 'User' and 'Dog-Owner' seperatly, what make the creation/deletion more complex.
3) Listen to a signal and manage all the creation from 'User'  - what seems to me the most ellegant way .

I have took my resource from https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html
Please let me know what you think @ormergi

Reply:
I find the "AbstractUser way" much more simple.

Reply:
Ok thanks for reviewing . @ormergi